src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -146,6 +146,9 @@ private TypeName getTypeNameForElementType(TypeRef type, boolean shouldBoxPrimit
     }
     switch (type.getKind()) {
       case TYPE_MESSAGE:
+        if (type.isMap() && type.isMessage() && type.isRepeated()) {
+          return getMapTypeName(type);
```

Comment:
Do we need something similar for `!type.isMap() && type.isRepeated()`?

Reply:
Actually if `isMap() == true`  then `isMessage() == true` and `isRepeated() == true` automatically (just checked the implementation of those method in api-compiler). Removed the other 2 conditions from the if statement as they are 100% redundant.